src/content/docs/products/ontrack/documentation/panopoto-integration/panopto-upload-process.md

```diff
@@ -0,0 +1,177 @@
+# Panopto Video Upload Process
+
+## Overview
+This document outlines key information about the process of uploading videos to Panopto using the Panopto REST API. While this isn't a comprehensive guide, it highlights the steps involved in creating a session, performing a multipart upload of the video file, creating the manifest, and finalising the upload. This serves as an informational resource for implementing the video upload functionality in the future. Additional research and context will be needed to fully integrate these steps.
+
+---
+
+## Step 1: Create a Session for Video Upload
+
+Before uploading a video, a session must be created. The session provides the necessary details for the upload process, including the upload target and session ID.
+
+###  Call the Session Creation API
+The session creation process is triggered by sending a POST request to the `sessionUpload` endpoint.
+
+- **API Endpoint:**
+  ```ruby
+  POST https://{server}/Panopto/PublicAPI/REST/sessionUpload
+  ```
+- **Payload:**
+  ```http
+  {
+    "FolderId": "{folder_id}"
+  }
+
+  ```
+- **Example Session Creation:**
+  ```ruby
+  response = RestClient.post(
+    "https://#{server}/Panopto/PublicAPI/REST/sessionUpload",
+    { "FolderId" => folder_id }.to_json,
+    { content_type: :json, accept: :json, authorization: "Bearer #{access_token}" }
+  )
+  ```
+
+- **Example Response:**
+  ```http
+  {
+    "ID": "session_id",
+    "UploadTarget": "upload_target_url",
+    "FolderId": "{folder_id}"
+  }
+  ```
+## Step 2: Upload Video via Multipart Upload
+Panopto supports multipart uploads for large video files. This step uploads the video in parts to the ```upload target``` specified in the session creation response.
+
+#### 1. Initiate Multipart Upload:
+- Get the Upload URL: After creating a session, extract the ```UploadTarget``` URL from the session creation response. This URL specifies where the video should be uploaded.
+
+- Set Up AWS SDK : Even though Panopto doesn’t require AWS credentials, use AWS SDK to handle the multipart upload. Initialise the SDK with dummy credentials (to bypass authentication but still use the multipart upload logic).
+
+```ruby
+# Extract the Upload Target URL from the session creation response
+upload_target = "https://deakin.au.panopto.com/Panopto/Upload/..."
+
+# Set up AWS SDK for multipart upload
+s3 = Aws::S3::Client.new(
+  endpoint: upload_target,
+  region: 'us-east-1',
+  access_key_id: 'dummy',
+  secret_access_key: 'dummy'
+)
+
+# Initiate multipart upload
+mpu = s3.create_multipart_upload(Bucket: 'upload-bucket', Key: 'video-file.mp4')
```

Comment:
Does the concept of bucket exist in Panopto or is this just a dummy too?
If Panopto offers storage via only S3 compatible object storage, then using the AWS SDK might make sense, but I doubt that is the case.

Reply:
Panopto uses S3-compatible object storage for uploading files. The concept of a bucket in Panopto is different from AWS S3. The S3 multipart upload method is how I chose to handle the file upload, but the actual storage is managed by Panopto, not AWS S3